src/provider.rs

```diff
@@ -15,14 +15,14 @@ pub trait Provider {
     /// See [RFC 6749, section 3.1](http://tools.ietf.org/html/rfc6749#section-3.1).
     ///
     /// Note: likely to become an associated constant.
-    fn auth_uri() -> &'static str;
+    fn auth_uri(&self) -> &str;
```

Comment:
Does this actually make sense for dynamic providers? I would guess they might want to return `String`, so maybe these should return `Cow`? (Or `Url` since that's what they're used for anyway)